### PR TITLE
[sc-10162] Move smartpay config from static to object

### PR DESCRIPTION
**Pull request by hSATAC**
Signed-off-by: Ash Wu <hsatac@gmail.com>

### CODE REVIEW

File: tests/SmartpayTest.php [hunk 1]
```diff
@@ -8,52 +8,38 @@
...
 
-    public function testSecretKeyAccessors()
+    public function testConstructor()
     {
-        Smartpay::setSecretKey(null);
-        static::assertNull(Smartpay::getSecretKey());
-
-        Smartpay::setSecretKey('sk_test_123');
-        static::assertSame('sk_test_123', Smartpay::getSecretKey());
+        $smartpay = new Smartpay('sec', 'pub');
```
Comment (othree) [inline tests/SmartpayTest.php:51]: Can we have a test case to have both ENV_VAR and the customer key as arguments when calling the constructor, and make sure the lib will use the later one?

Comment (hSATAC) [reply tests/SmartpayTest.php:51]: Added

File: composer.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
     "guzzlehttp/guzzle": ">=6.5",
     "ircmaxell/random-lib": "^1.2",
     "caseyamcl/guzzle_retry_middleware": "^2.2",
-    "tuupola/base62": ">= 1.0"
+    "tuupola/base62": ">= 1.0",
+    "ext-json": "*"
```
Comment (othree) [inline composer.json:22]: Do we use this dependency? I can't find it. (Sorry I forgot to submit this comment)

Comment (hSATAC) [reply composer.json:22]: Short answer: it's for the built-in `json_decode` function we used in our code base. Some PHP installations could have been built without json extension.

Long answer: See this [article](https://luis-barros-nobrega.medium.com/ext-json-is-missing-in-composer-json-what-is-this-700a1fcad966) for detailed explanation.

**Review by othree — APPROVED** [review]

**Pull request merged**